apps/cli/src/runAstgrepCodemod.ts

```diff
@@ -0,0 +1,62 @@
+import fs from 'fs';
+import { execSync } from 'node:child_process';
+import { readFile } from 'node:fs/promises';
+import path from 'path';
+import * as yaml from 'js-yaml';
+import { PrinterBlueprint } from './printer.js';
+
+export const runAstgrep = async (
+	printer: PrinterBlueprint,
+	rulePath: string,
+	targetDirectory: string,
+): Promise<void> => {
+	const yamlString = await readFile(rulePath, { encoding: 'utf8' });
+	const yamlObject = yaml.load(yamlString);
+	const language = languageToExtension(yamlObject.language);
+	printer.printConsoleMessage(
+		'info',
+		`Executing ast-grep for language : ${language}`,
+	);
+
+	// Function to recursively iterate over files in the directory
+	const iterateFiles = async (dirPath: string) => {
+		const entries = await fs.promises.readdir(dirPath, {
+			withFileTypes: true,
+		});
+
+		// Iterate over each entry in the directory
+		for (const entry of entries) {
+			const entryPath = path.join(dirPath, entry.name);
+			// Check if the entry is a directory
+			if (entry.isDirectory()) {
+				// Recursively call the function for subdirectories
+				await iterateFiles(entryPath);
+			} else if (entry.isFile()) {
+				// If the entry is a file, log its path
+				const fileExtension = path.extname(entryPath).slice(1);
+				if (fileExtension !== language) {
+					continue;
+				}
+				const astCommand = `sg scan -r ${rulePath} ${entryPath} -U`;
+				execSync(astCommand);
```

Comment:
i'm not sure this is going to work unless user has installed ast-grep on his machine

Reply:
also we don't need to execute changes here, we need to build new files, then execute buildFileCommands() and pass it up. similar how it's done for other engines. not sure if you're doing the same here.

Reply:
I think just installing it for the user is a good idea to get ahead. Also this ll be in beta mode, so we can work on better devX as we move ahead. 

What do you mean by build new files.